gsy_framework/utils.py

```diff
@@ -585,3 +590,36 @@ def use_default_if_null(input_value: Union[int, str, float], default_value: Unio
                         ) -> Union[int, str, float]:
     """Return input_value if not null, else return provided default_value"""
     return input_value if input_value not in NULL_VALUES else default_value
+
+
+def _generate_time_slot_list(slot_length: timedelta, sim_duration: timedelta,
+                             start_date: datetime) -> List[datetime]:
+    return [start_date.add(minutes=slot_length.seconds / 60 * time_diff) for time_diff in
+            range(int(sim_duration.total_seconds() / slot_length.total_seconds()))]
+
+
+def resample_hourly_energy_profile(
```

Comment:
I think it is a better fit to live with the other profile methods of gsy-framework, therefore IMO read_user_profile is a much better module for the function